test/set/pathValues/atom.spec.js

```diff
@@ -167,6 +167,25 @@ describe("an atom", function() {
         }));
     });
 
+    it("through a reference that lands on an atom with a null last key", function() {
+
+        var cache = {};
+        var version = 0;
+        setPathValues(
+            getModel({ cache: cache, version: version++ }), [
+                $pathValue("foo.value", $ref("bar.value")),
+                $pathValue("bar.value", $atom("bar-value")),
+                $pathValue(["foo", "value", null], $atom("foo-value-1")),
```

Comment:
@sdesai the first set to `["foo", "value", null]` is ensuring the hard-link is assigned from `["foo", "value"]` to `["bar", "value"]`. `setPathValues` is writing the first two PathValues directly, but the hard-link is only set up once the reference (`foo.value`) is crossed during a get or set. I wrote two distinct values ('foo-value-1' and 'foo-value-2') to validate that the second value was written by crossing the hard-link.